### PR TITLE
Add YOLOv10 Object Detection Support

### PR DESCRIPTION
**Pull request by NickHerrig**
# Description

This PR adds support for Yolov10 Object Detection models.

List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [X] New feature (non-breaking change which adds functionality)
-   [X] This change requires a documentation update

## How has this change been tested, please provide a testcase or example of how you tested the change?

TBD

## Docs

-   [ ] TBD

### CODE REVIEW

File: inference/models/utils.py [hunk 1]
```diff
@@ -50,6 +51,13 @@
     ("object-detection", "yolo_nas_s"): YOLONASObjectDetection,
     ("object-detection", "yolo_nas_m"): YOLONASObjectDetection,
     ("object-detection", "yolo_nas_l"): YOLONASObjectDetection,
+    ("object-detection", "yolov10"): YOLOv10ObjectDetection,
+    ("object-detection", "yolov10s"): YOLOv10ObjectDetection,
+    ("object-detection", "yolov10n"): YOLOv10ObjectDetection,
+    ("object-detection", "yolov10s"): YOLOv10ObjectDetection,
```
Comment (NickHerrig) [inline inference/models/utils.py:57]: should be b

File: inference/models/yolov10/yolov10_object_detection.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+from inference.core.models.object_detection_base import (
+    ObjectDetectionBaseOnnxRoboflowInferenceModel,
+)
+
+
+class YOLOv10ObjectDetection(ObjectDetectionBaseOnnxRoboflowInferenceModel):
+    """Roboflow ONNX Object detection model (Implements an object detection specific infer method).
+
+    This class is responsible for performing object detection using the YOLOv8 model
```
Comment (NickHerrig) [inline inference/models/yolov10/yolov10_object_detection.py:13]: change to yolov10

**Review by probicheaux — APPROVED** [review]
lgtm

**Review by probicheaux — APPROVED** [review]
lgtm2

**Pull request merged**